velox/dwio/parquet/thrift/ThriftTransport.h

```diff
@@ -21,12 +21,69 @@
 
 namespace facebook::velox::parquet::thrift {
 
-class ThriftBufferedTransport
-    : public apache::thrift::transport::TVirtualTransport<
-          ThriftBufferedTransport> {
+class ThriftTransport
+    : public apache::thrift::transport::TVirtualTransport<ThriftTransport> {
+ public:
+  virtual uint32_t read(uint8_t* outputBuf, uint32_t len) = 0;
```

Comment:
No need to use two transports. The footer can also use the steaming one. This will remove one memory copy for the footer as well.
Can you also run the microbenchmark to show the before and after in the PR message?
There are also some minor comments I'll leave tomorrow.
Cc @Yuhta We will create BufferedSeekableImputStream to hide bufferStart and bufferEnd in multiple places and separate the concepts of buffer management and parquet business logic after this PR.

Reply:
`ThriftStreamingTransport` is not sufficient for footer. This is how the footer is read: 
At beginning, we take a guess on the footer size, which is at least enough for the footer length and `PAR1`. And then, 1) if the guessed range is sufficient, we use it. Otherwise, 2) we use the actual footer length and re-load again.

In this case, at least in situation 1), we need `ThriftBufferedTransport` parse the footer, because it is guessed and the actually start point of the footer might be after the guessed point.

For situation 2), we can have another PR to optimize and use `ThriftStreamingTransport`

Reply:
This PR does not change the current page header reading and footer reading logic. This is a prerequisite PR. So the benchmark should remain the same so far.

Reply:
@yingsu00 Yes it's a good idea to encapsulate the range management of `SeekableImputStream` in a common place.  I would suggest the composition pattern (i.e. wrap `SeekableImputStream` in a helper class along with the buffer) instead of inherit it.  Composition is always easier for future refactoring than inheritance, unless you have a lot delegations to do.